src/objects.rs

```diff
@@ -345,6 +345,21 @@ impl Id<StopArea> for StopPoint {
 }
 impl_codes!(StopPoint);
 
+pub type Date = String;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum ExceptionType {
+    #[serde(rename = "1")] Add,
+    #[serde(rename = "2")] Remove,
+}
+
+pub type CalendarDatesT = Vec<(Date, ExceptionType)>;
+
+pub trait CalendarDates {
+    fn calendar_dates(&self) -> &CalendarDatesT;
+    fn calendar_dates_mut(&mut self) -> &mut CalendarDatesT;
+}
```

Comment:
You don't need this trait machinery. It is needed for codes because several objects have codes, but for callendar_dates, only calendars will be impacted.

Thus you can just have `pub type CalendarDates = Vec<(Date, ExceptionType)>;` and then just directly take a Calendar object.

Reply:
i'm a robot :)